li-apache-kafka-clients/src/main/java/com/linkedin/kafka/clients/consumer/LiKafkaFederatedConsumerImpl.java

```diff
@@ -325,82 +313,101 @@ public void unsubscribe() {
 
   @Override
   public void commitSync() {
-    throw new UnsupportedOperationException("Not implemented yet");
+    commitSync(Duration.ofMillis(_defaultApiTimeoutMs));
   }
 
   @Override
   public void commitSync(Duration timeout) {
-    throw new UnsupportedOperationException("Not implemented yet");
+    executeCallbackWithTimeout("commitSync", null,
+        (consumer, ignored, timeoutDuration) -> consumer.commitSync(timeoutDuration), timeout);
   }
 
   @Override
   public void commitSync(Map<TopicPartition, OffsetAndMetadata> offsets) {
-    throw new UnsupportedOperationException("Not implemented yet");
+    commitSync(offsets, Duration.ofMillis(_defaultApiTimeoutMs));
   }
 
   @Override
   public void commitSync(Map<TopicPartition, OffsetAndMetadata> offsets, Duration timeout) {
-    throw new UnsupportedOperationException("Not implemented yet");
+   executeCallbackWithTimeout("commitSync", getPartitionValueMapByCluster(offsets),
+        (consumer, perClusterOffsets, timeoutDuration) -> {
+      consumer.commitSync((Map<TopicPartition, OffsetAndMetadata>) perClusterOffsets, timeoutDuration);
+        }, timeout);
   }
 
   @Override
   public void commitAsync() {
-    throw new UnsupportedOperationException("Not implemented yet");
+    commitAsync(null);
   }
 
   @Override
   public void commitAsync(OffsetCommitCallback callback) {
-    throw new UnsupportedOperationException("Not implemented yet");
+    for (ClusterConsumerPair<K, V> entry : getImmutableConsumerList()) {
+      entry.getConsumer().commitAsync(callback);
```

Comment:
this would call a user-provided callback N times for N consumers. we need to aggregate them 1st and call the user callback only once

Reply:
Thanks for catching this. Added a wrapper callback, which will call the user callback once.